src/__mocks__/zustand.js

```diff
@@ -5,14 +5,23 @@ const { act } = require('react-dom/test-utils');
 const storeResetFns = new Set();
 
 const create = (createState) => {
-  const store = actualCreate(createState);
-
   /**
-   * if createState is empty, store won't have a getState function
-   * then we need to skip associating this value
+   * If createState is empty, store won't have a getState function
+   * and it will be the real createImpl function.
+   *
+   * In this situation, we need to make sure to return the function
+   * itself to be able to intercept the original createState function
+   * and be able to reset the store after each test.
+   *
    * This happens whenever we use the following syntax
    * create()(...) to help TS auto completion
    */
+  if (createState === undefined) {
+    return create;
```

Comment:
This is the point when we need to return the modified `createState` function, to avoid returning the real one.